### PR TITLE
 fix #28926 pandas\api\test_api.py mypy errors

### PR DESCRIPTION
**Pull request by yogendrasoni**
- [ ] xref  #28926 
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/api/test_api.py [hunk 1]
```diff
@@ -41,7 +41,8 @@ class TestPDApi(Base):
     ]
 
     # these are already deprecated; awaiting removal
-    deprecated_modules = []
+    # GH #28926
```
Comment (WillAyd) [inline pandas/tests/api/test_api.py]: Thanks for the thoroughness but for this change don't need the GH issue references

File: pandas/tests/api/test_api.py [hunk 2]
```diff
@@ -41,7 +41,8 @@ class TestPDApi(Base):
...
+    deprecated_modules : List[str] = []
```
Comment (WillAyd) [inline pandas/tests/api/test_api.py]: Right now on master we still have 3.5 in our CI, so need to use the `# type: ...` comment syntax for compat

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/tests/api/test_api.py (hunk 2 above)
Comment (yogendrasoni) [reply pandas/tests/api/test_api.py]: fixed comment syntax. passed black and flake8. Do i need to remove GH # reference?

Comment (WillAyd) [reply pandas/tests/api/test_api.py]: Yes go ahead and remove

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]
@yogendrasoni can you also remove the `ignore_errors` entry for this module from setup.cfg.

File: pandas/tests/api/test_api.py (hunk 2 above)
Comment (yogendrasoni) [reply pandas/tests/api/test_api.py]: done

File: setup.cfg [hunk 3]
```diff
@@ -134,7 +134,7 @@ no_implicit_optional=True
 ignore_errors=True
 
 [mypy-pandas.tests.api.test_api]
-ignore_errors=True
+ignore_errors=False
```
Comment (simonjayhawkins) [inline setup.cfg]: can just remove entirely as the default for ignore_errors is False

Comment (yogendrasoni) [reply setup.cfg]: do i need to remove entire section
 - [mypy-pandas.tests.api.test_api]
 - ignore_errors=True

Comment (simonjayhawkins) [reply setup.cfg]: yes

Comment (yogendrasoni) [reply setup.cfg]: done

File: pandas/tests/api/test_api.py [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import pandas as pd
 from pandas import api, compat
 from pandas.util import testing as tm
+from typing import List
```
Comment (simonjayhawkins) [inline pandas/tests/api/test_api.py]: you need to sort imports https://dev.pandas.io/docs/development/contributing.html#import-formatting

File: setup.cfg (hunk 3 above)
Comment (simonjayhawkins) [reply setup.cfg]: remove L134-136

File: setup.cfg [hunk 5]
```diff
@@ -131,10 +131,9 @@ ignore_missing_imports=True
 no_implicit_optional=True
 
 [mypy-pandas.conftest]
-ignore_errors=True
```
Comment (simonjayhawkins) [inline setup.cfg]: revert this deletion

File: pandas/tests/api/test_api.py [hunk 6]
```diff
@@ -41,7 +42,7 @@ class TestPDApi(Base):
     ]
 
     # these are already deprecated; awaiting removal
-    deprecated_modules = []
+    deprecated_modules = []  # type: List[int]
```
Comment (simonjayhawkins) [inline pandas/tests/api/test_api.py]: why is this List[int], was List[str]?

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]

**Review by simonjayhawkins — APPROVED** [review]
@yogendrasoni lgtm @WillAyd

**Comment by yogendrasoni** [comment]
> @yogendrasoni lgtm @WillAyd

@WillAyd  , @simonjayhawkins simonjayhawkins anything else is required ?

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks!

**Pull request merged**